src/syscheckd/syscheck_audit.c

```diff
@@ -274,6 +273,24 @@ int add_audit_rules_syscheck(bool first_time) {
 }
 
 
+void audit_no_rules_to_realtime() {
+    int found;
+    int i;
+
+    for (i = 0; syscheck.dir[i] != NULL; i++) {
+        if (syscheck.opts[i] & WHODATA_ACTIVE) {
+            found = search_audit_rule(fim_get_real_path(i), "wa", AUDIT_KEY);
+
+            if (found == 0) {   // No rule found
+                mwarn(FIM_ERROR_WHODATA_ADD_DIRECTORY, fim_get_real_path(i));
+                syscheck.opts[i] &= ~WHODATA_ACTIVE;
+                syscheck.opts[i] |= REALTIME_ACTIVE;
+            }
+        }
+    }
```

Comment:
The content of this for loop could be rewritten so it has one less level of indentation by using a `continue` with the inverted condition of the first if, like this:
```c
    for (i = 0; syscheck.dir[i] != NULL; i++) {
        if ((syscheck.opts[i] & WHODATA_ACTIVE) == 0) {
            continue;
        }

        found = search_audit_rule(fim_get_real_path(i), "wa", AUDIT_KEY);

        if (found == 0) {   // No rule found
            mwarn(FIM_ERROR_WHODATA_ADD_DIRECTORY, fim_get_real_path(i));
            syscheck.opts[i] &= ~WHODATA_ACTIVE;
            syscheck.opts[i] |= REALTIME_ACTIVE;
        }
    }
```